### DISCUSSION TITLE
List SFTP directory with libcurl

### DISCUSSION THREAD

**Discussion by bschaer** [curl/curl#6801] [Q&A]
Is it somehow possible to process a list of directory entries with SFTP without having to parse the entries from a text buffer?
If yes, I didn't find out how...

**Comment by dfandrich** [comment]
The best place for questions about the curl API is the curl-library mailing
list, not here. But, have you tried the CURLOPT_DIRLISTONLY option?

**Reply by bschaer** [reply]
The problem is not that I can't list the content. But it's actually written with the write callback funtion registered with _CURLOPT_WRITEFUNCTION_. And from there I have to parse them manually which is not such an easy task, because spaces are used as field separators and at the same time can be part of file names.

And what I learned from the SFTP RFCs is, that it would be all available internally in non-textual form since SFTP uses a binary kind of protocol. So I think it's quite a bit cumbersome to parse it if it has already been available as separate fields.

I actually made same experiments with the FTP options **CURLOPT_WILDCARDMATCH** in conjunction with **CURLOPT_CHUNK_BGN_FUNCTION**. And what is passed to this callback function is exactly the kind of information I wanted. But it seems it's just working on FTP, and not on SFTP.

**Reply by bagder** [reply]
`CURLOPT_WILDCARDMATCH` could indeed be made to work with SFTP and I think it could work pretty smoothly. It just requires that someone rolls up their sleeves and get to work... :wink:
